projects/aca-content/src/lib/components/search/search-input.service.ts

```diff
@@ -23,16 +23,22 @@
  * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
  */
 
-import { NgModule } from '@angular/core';
-import { CommonModule } from '@angular/common';
-import { CoreModule } from '@alfresco/adf-core';
-import { AppHeaderComponent } from './header.component';
-import { AppSearchInputModule } from '../search/search-input.module';
-import { AppToolbarModule } from '../toolbar/toolbar.module';
+import { Injectable } from '@angular/core';
+import { Router } from '@angular/router';
 
-@NgModule({
-  imports: [CommonModule, CoreModule.forChild(), AppSearchInputModule, AppToolbarModule],
-  declarations: [AppHeaderComponent],
-  exports: [AppHeaderComponent]
+@Injectable({
+  providedIn: 'root'
 })
-export class AppHeaderModule {}
+export class SearchInputService {
+  savedRoute: '';
+
+  constructor(private router: Router) {}
+
+  saveRoute(route) {
+    this.savedRoute = route;
+  }
+
+  exitSearch() {
+    this.router.navigate([this.savedRoute]);
```

Comment:
before navigation, should be testing if the route is defined; and you should have 2 unit test cases: one when route is defined, and another is when it's not